onnx/backend/test/case/node/convtranspose.py

```diff
@@ -279,3 +279,51 @@ def export_convtranspose_pads():  # type: () -> None
                         [13., 7., 15.]]]]).astype(np.float32)
 
         expect(node, inputs=[x, W], outputs=[y], name='test_convtranspose_pads')
+
+    @staticmethod
+    def export_with_kernel():  # type: () -> None
+
+        node = onnx.helper.make_node(
+            'ConvTranspose', ['x', 'w'], ['y'],
+            name='test',
+            strides=[3, 2],
+            output_shape=[10, 8],
+            kernel_shape=[3, 3],
+            output_padding=[1, 1]
+        )
+
+        x = np.array([[[[0., 1., 2.],  # (1, 1, 3, 3)
+                        [3., 4., 5.],
+                        [6., 7., 8.]]]]).astype(np.float32)
+
+        W = np.array([[[[1., 1., 1.],  # (1, 2, 3, 3)
+                        [1., 1., 1.],
+                        [1., 1., 1.]],
+                       [[1., 1., 1.],
+                        [1., 1., 1.],
+                        [1., 1., 1.]]]]).astype(np.float32)
+
+        y = np.array([[[[0., 0., 1., 1., 3., 2., 2., 0.],  # (1, 2, 10, 8)
+                        [0., 0., 1., 1., 3., 2., 2., 0.],
+                        [0., 0., 1., 1., 3., 2., 2., 0.],
+                        [3., 3., 7., 4., 9., 5., 5., 0.],
+                        [3., 3., 7., 4., 9., 5., 5., 0.],
+                        [3., 3., 7., 4., 9., 5., 5., 0.],
+                        [6., 6., 13., 7., 15., 8., 8., 0.],
+                        [6., 6., 13., 7., 15., 8., 8., 0.],
+                        [6., 6., 13., 7., 15., 8., 8., 0.],
+                        [0., 0., 0., 0., 0., 0., 0., 0.]],
+
+                       [[0., 0., 1., 1., 3., 2., 2., 0.],
+                        [0., 0., 1., 1., 3., 2., 2., 0.],
+                        [0., 0., 1., 1., 3., 2., 2., 0.],
+                        [3., 3., 7., 4., 9., 5., 5., 0.],
+                        [3., 3., 7., 4., 9., 5., 5., 0.],
+                        [3., 3., 7., 4., 9., 5., 5., 0.],
+                        [6., 6., 13., 7., 15., 8., 8., 0.],
+                        [6., 6., 13., 7., 15., 8., 8., 0.],
+                        [6., 6., 13., 7., 15., 8., 8., 0.],
+                        [0., 0., 0., 0., 0., 0., 0., 0.]]]]).astype(np.float32)
+
+        expect(node, inputs=[x, W], outputs=[y],
+               name='test_convtranspose_with_kernel')
```

Comment:
you need to add kernel_shape attribute to the expect call here (kernel_shape=`W.shape`)

Reply:
actually you can just add an additional `expect` call to existing test cases (with kernel_shape attribute provided), so we can avoid copy-paste these pre-computed reference inputs & outputs.

Reply:
Yes I was thinking if it is a better way...

Reply:
@bddppq btw I wonder why we need to add "kernel_shape = W.shape" here, since I have already added it to the node attribute.

Reply:
@zrphercule you are right, it should be in the node.